tests/test_harvester_daily_bfg_tmp2m.py

```diff
@@ -0,0 +1,200 @@
+#!/usr/bin/env python
+
+import os
+import sys
+from pathlib import Path 
+
+import numpy as np
+from datetime import datetime
+import pytest
+import yaml
+from netCDF4 import Dataset
+
+from score_hv import hv_registry
+from score_hv.harvester_base import harvest
+from score_hv.yaml_utils import YamlLoader
+from score_hv.harvesters.innov_netcdf import Region, InnovStatsCfg
+
+TEST_DATA_FILE_NAMES = ['tmp2m_bfg_2023032100_fhr09_control.nc',
+                        'tmp2m_bfg_2023032106_fhr06_control.nc',
+                        'tmp2m_bfg_2023032106_fhr09_control.nc',
+                        'tmp2m_bfg_2023032112_fhr06_control.nc',
+                        'tmp2m_bfg_2023032112_fhr09_control.nc',
+                        'tmp2m_bfg_2023032118_fhr06_control.nc',
+                        'tmp2m_bfg_2023032118_fhr09_control.nc',
+                        'tmp2m_bfg_2023032200_fhr06_control.nc']
+
+DATA_DIR = os.path.join(Path(__file__).parent.parent.resolve(), 'data')
+GRIDCELL_AREA_DATA_PATH = os.path.join(DATA_DIR,
+                                       'gridcell-area' + 
+                                       '_noaa-ufs-gefsv13replay-pds' + 
+                                       '_bfg_control_1536x768_20231116.nc')
+
+CONFIGS_DIR = 'configs'
+PYTEST_CALLING_DIR = Path(__file__).parent.resolve()
+TEST_DATA_PATH = os.path.join(PYTEST_CALLING_DIR, 'data')
+BFG_PATH = [os.path.join(TEST_DATA_PATH,
+                         file_name) for file_name in TEST_DATA_FILE_NAMES]
+
+VALID_CONFIG_DICT = {'harvester_name': hv_registry.DAILY_BFG,
+                     'filenames' : BFG_PATH,
+                     'statistic': ['mean', 'variance', 'minimum', 'maximum'],
+                     'variable': ['tmp2m']}
+
+def test_variable_names():
+    data1 = harvest(VALID_CONFIG_DICT)
+    assert data1[0].variable == 'tmp2m'
+
+def test_global_mean_values(tolerance=0.001):
+    """The value of 287.0713362523281 is the mean value of the global means
+    calculated from eight forecast files:
+        
+        tmp2m_bfg_2023032100_fhr09_control.nc
+        tmp2m_bfg_2023032106_fhr06_control.nc
+        tmp2m_bfg_2023032106_fhr09_control.nc
+        tmp2m_bfg_2023032112_fhr06_control.nc
+        tmp2m_bfg_2023032112_fhr09_control.nc
+        tmp2m_bfg_2023032118_fhr06_control.nc
+        tmp2m_bfg_2023032118_fhr09_control.nc
+        tmp2m_bfg_2023032200_fhr06_control.nc
+        
+    When averaged together, these files represent a 24 hour mean. The average 
+    value hard-coded in this test was calculated from these forecast files 
+    using a separate python code.
+    """
+    data1 = harvest(VALID_CONFIG_DICT)
+    global_mean = 287.0713362523281
+    assert data1[0].value <= (1 + tolerance) * global_mean
+    assert data1[0].value >= (1 - tolerance) * global_mean 
+
+def test_global_mean_values2(tolerance=0.001):
```

Comment:
Similar comment to the OLR PR, I believe that best practice for clarity and understanding would be to give this test a more unique distinction from the other global mean test than "2" as it isn't clear what the main difference in the tests is from the name, there's no comment, and the tests are actually operating fairly differently so warrant a distinguishing set of names.

Reply:
changed to test_global_mean_values_offline() and test_global_mean_values_netCDF4()